File: src/main/java/com/wowo/wowo/domain/shared/DomainEvent.java
package com.wowo.wowo.domain.shared;

import java.time.Instant;

/**
 * Base interface for all domain events in the system.
 * Domain events represent something important that happened in the domain.
 */
public interface DomainEvent {
    
    /**
     * Gets the timestamp when this event occurred.
     */
    Instant getOccurredOn();
    
    /**
     * Gets a unique identifier for this event.
     */
    String getEventId();
    
    /**
     * Gets the type of this event.
     */
    String getEventType();
}

File: src/main/kotlin/com/wowo/wowo/shared/domain/DomainEvent.kt
package com.wowo.wowo.shared.domain

import java.time.LocalDateTime
import java.util.*

/**
 * Base interface for all domain events
 */
interface DomainEvent {
    val eventId: String
    val occurredOn: LocalDateTime
    val aggregateId: String
}

/**
 * Abstract base implementation for domain events
 */
abstract class BaseDomainEvent(
    override val aggregateId: String
) : DomainEvent {
    override val eventId: String = UUID.randomUUID().toString()
    override val occurredOn: LocalDateTime = LocalDateTime.now()
}

